### ISSUE TITLE
test_collect_invalid_signature_message fails with Python 2.7

### ISSUE THREAD

**Issue by kulikjak** [pytest-dev/pytest#4815]
One test introduced in #4196 (Issue 4026) fails for me with Python 2.7.

    import pytest
    class TestCase:
        @pytest.fixture(autouse=True)
        def fix():
            pass

Instead of 

    Could not determine arguments of ....

it ends up with

    collected 0 items
    === no tests ran in 0.01 seconds ===

and without any error. Python 3 versions do fail as expected.

Here is some info about the system:

    platform sunos5 -- Python 2.7.15, pytest-4.3.0, py-1.7.0, pluggy-0.8.1
    hypothesis profile 'default' -> database=DirectoryBasedExampleDatabase('...')
    rootdir: .../pytest, inifile:
    plugins: hypothesis-4.7.0

**Comment by asottile** [comment]
I can't reproduce myself:

```console
$ pytest t.py 
============================= test session starts ==============================
platform linux2 -- Python 2.7.15rc1, pytest-4.3.0, py-1.8.0, pluggy-0.8.1
rootdir: /tmp/t, inifile:
collected 0 items / 1 errors                                                   

==================================== ERRORS ====================================
____________________________ ERROR collecting t.py _____________________________
Could not determine arguments of <bound method ?.fix of <t.TestCase instance at 0x7f6b88cbe8c0>>: invalid method signature
!!!!!!!!!!!!!!!!!!! Interrupted: 1 errors during collection !!!!!!!!!!!!!!!!!!!!
=========================== 1 error in 0.07 seconds ============================
$ pip install 'py<1.8'
...
$ pytest t.py 
============================= test session starts ==============================
platform linux2 -- Python 2.7.15rc1, pytest-4.3.0, py-1.7.0, pluggy-0.8.1
rootdir: /tmp/t, inifile:
collected 0 items / 1 errors                                                   

==================================== ERRORS ====================================
____________________________ ERROR collecting t.py _____________________________
Could not determine arguments of <bound method ?.fix of <t.TestCase instance at 0x7f6c9b4572d8>>: invalid method signature
!!!!!!!!!!!!!!!!!!! Interrupted: 1 errors during collection !!!!!!!!!!!!!!!!!!!!
=========================== 1 error in 0.04 seconds ============================
```

Do you have some more information on how we can reproduce what you're seeing?  What's the output of `pip freeze --all`?

**Comment by kulikjak** [comment]
Here it is:

    anyjson==0.3.3
    asn1crypto==0.24.0
    astroid==1.4.8
    atomicwrites==1.3.0
    attrs==18.2.0
    Babel==2.3.4
    cffi==1.11.4
    CherryPy==5.1.0
    cmd2==0.6.8
    coverage==4.5.2
    cryptography==2.1.4
    debtcollector==1.3.0
    dnet==1.12
    dnspython==1.12.0
    enum34==1.1.6
    eventlet==0.18.4
    filechunkio==1.6
    FormEncode==1.3.0
    funcsigs==0.4
    futures==3.0.5
    greenlet==0.4.9
    hypothesis==4.7.0
    idna==2.0
    iniparse==0.4
    ipaddress==1.0.16
    ipython==0.10
    iso8601==0.1.11
    isort==4.2.5
    jsonrpclib-pelix==0.2.6
    jsonschema==2.0.0
    keystoneauth1==2.4.0
    lazy-object-proxy==1.2.2
    logilab-common==1.2.2
    lxml==3.6.4
    M2Crypto==0.24.0
    Mako==1.0.4
    MarkupSafe==0.23
    mccabe==0.5.2
    mercurial==4.7.1
    mock==1.3.0
    monotonic==0.6
    more-itertools==5.0.0
    msgpack-python==0.4.7
    MySQL-python==1.2.5
    netaddr==0.7.18
    netifaces==0.10.4
    netsnmp-python==1.0a1
    nose==1.3.7
    numpy==1.14.2
    oauthlib==1.0.3
    oslo.config==3.9.0
    oslo.i18n==3.5.0
    oslo.serialization==2.4.0
    oslo.utils==3.8.0
    osprofiler==1.2.0
    Paste==2.0.3
    PasteDeploy==1.5.2
    pathlib2==2.3.3
    pbr==1.8.1
    Pillow==5.1.0
    pip==10.0.1
    pluggy==0.8.1
    ply==3.8
    portend==2.3
    positional==1.0.1
    prettytable==0.7.2
    py==1.7.0
    pyasn1==0.4.2
    pyasn1-modules==0.2.1
    pybonjour==1.1.1
    pycparser==2.14
    pycurl==7.43.0
    pyflakes==0.8.1
    Pygments==2.1.3
    pylint==1.6.4
    pyOpenSSL==16.2.0
    pyparsing==2.1.4
    pytest==4.3.0
    python-ceilometerclient==2.4.0
    python-keystoneclient==2.3.1
    python-ldap==2.4.25
    python-swiftclient==3.0.0
    pytz==2016.4
    repoze.lru==0.6
    requests==2.9.1
    scandir==1.9.0
    scons==2.5.1
    setuptools==39.2.0
    setuptools-scm==2.1.0
    simplejson==3.6.5
    six==1.10.0
    SQLAlchemy==1.0.12
    stevedore==1.12.0
    tempora==1.12
    Twisted==10.1.0
    urllib3==1.15.1
    WebOb==1.5.1
    wrapt==1.10.6
    zope.interface==4.6.0

**Comment by nicoddemus** [comment]
@kulikjak unfortunately I cannot fully reproduce that environment on Windows, had to comment out a few dependencies:

* cryptography==2.1.4
* dnet==1.12
* idna==2.0
* ipython==0.10
* lxml==3.6.4
* M2Crypto==0.24.0
* MySQL-python==1.2.5
* netsnmp-python==1.0a1
* pybonjour==1.1.1

But those do not seem related to the problem.

Anyway I also cannot reproduce the issue myself.

**Comment by kulikjak** [comment]
Well, that's what I was afraid of. It might even be some Solaris related problem.

Can somebody, please, point me to a place in the code (or file) of pytest where this should happen? I will try to find it and let you know.

**Comment by asottile** [comment]
This is the place where the exception (and eventual error message) is generated: https://github.com/pytest-dev/pytest/pull/4196/files#diff-4ec37b1ad5b7d41fdac56c907d08dde1R137

**Comment by kulikjak** [comment]
I found the problem.

Upgrading to the newest version of `funcsigs` library solved the issue. I tried other versions and it looks fixed since version `1.0.0`. Setup.py should thus be updated to (?):

`'funcsigs>=1.0.0;python_version<"3.0"'`
